Submissions/Bain_Week 3.playground/Contents.swift

```diff
@@ -0,0 +1,227 @@
+        self.sideLength = sideLength
+        super.init(name: name)
+        numberOfSides = 4
+    }
+    
+    func area() -> Double {
+        return sideLength * sideLength
+    }
+    
+    override func simpleDescription() -> String {
+        return "A square with sides of length \(sideLength)."
+    }
+}
+let test = Square(sideLength: 5.2, name: "my test square")
+test.area()
+test.simpleDescription()
+
+class Circle: NamedShape {
+    var radius: Double
+    
+    init(radius: Double, name: String) {
+        self.radius = radius
+        super.init(name: name)
+    }
+    
+    func area() -> Double {
+        return 3.14 * radius * radius
+    }
+    
+    override func simpleDescription() -> String {
+        return "A circle with the radius of \(radius)"
+    }
+}
+
+let test2 = Circle(radius: 4, name: "my test circle")
+test2.area()
+test2.simpleDescription()
+
+//: ## Enumerations and Structures
+//: 6.)
+//:
+//: Write a function that compares two `Rank` values by comparing their raw values.
+//:
+enum Rank: Int {
+    case ace = 1
+    case two, three, four, five, six, seven, eight, nine, ten
+    case jack, queen, king
+    
+    func simpleDescription() -> String {
+        switch self {
+        case .ace:
+            return "ace"
+        case .jack:
+            return "jack"
+        case .queen:
+            return "queen"
+        case .king:
+            return "king"
+        default:
+            return String(self.rawValue)
+        }
+    }
+}
+let ace = Rank.ace
+let aceRawValue = ace.rawValue
+
+//: 7.)
+//:
+//: Add a `color()` method to `Suit` that returns “black” for spades and clubs, and returns “red” for hearts and diamonds.
+//:
+enum Suit: CaseIterable {
+    case spades, hearts, diamonds, clubs
+    
+    func simpleDescription() -> String {
+        switch self {
+        case .spades:
+            return "spades"
+        case .hearts:
+            return "hearts"
+        case .diamonds:
+            return "diamonds"
+        case .clubs:
+            return "clubs"
+        }
+    }
+    
+    func color(suit: Suit) -> String {
+        if suit == .spades || suit == .clubs {
+            return "black"
+        } else {
+            return "red"
+        }
+    }
+}
+let hearts = Suit.hearts
+let heartsDescription = hearts.simpleDescription()
+let heartsColor = hearts.color(suit: hearts)
+
+//: 8.)
+//:
+//: Add a third case to `ServerResponse` and to the switch.
+//:
+enum ServerResponse {
+    case result(String, String)
+    case failure(String)
+    case guest(String)
+}
+
+let success = ServerResponse.result("6:00 am", "8:09 pm")
+let failure = ServerResponse.failure("Out of cheese.")
+let partyOfTwo = ServerResponse.guest("Another order of cheese is on its way!")
+
+switch success {
+case let .result(sunrise, sunset):
+    print("Sunrise is at \(sunrise) and sunset is at \(sunset).")
+case let .failure(message):
+    print("Failure...  \(message)")
+case let .guest(name):
```

Comment:
the variable inside the parentheses is bound to the scope inside the `case let` - this is why you have a warning telling you that `Immutable value 'name' was never used; consider replacing with '_' or removing it'`

Your .guest case actually prints "Enjoy your dinner for two. guest("Another order of cheese is on its way!"..." because `partyOfTwo` is a variable referring to ServerResponse instance. You wanted to use the `name` variable you bound in the `case let`, and perhaps declare partyOfTwo's initializing string as someone's name to be consistent in your example. :-)